### DISCUSSION TITLE
git push issue

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#6068] [General]
Hi

Occasionally `git push` will stop working, and I'm not entirely sure why?

### Setup

- My site is hosted on Digital Ocean and I'm using Ploi to mange it
- I'm using Redis, Horizon and an artisan horizon daemon process
- Statamic Git integration has been configured, and when it works, it's a brilliant workflow
- My deploy script ends with this `php artisan horizon:terminate`

### The problem

`git push` sometimes stops working?  I `ssh` into the staging server and I run `git status` I see this message

```
Your branch is ahead of 'origin/staging' by X commits.
```

If I try to manually `git push` I see this message

```
remote: Invalid credentials
fatal: Authentication failed for 'https:..
```

Even though the credentials usually do work… Statamic happily pushes commits to `origin/staging` when it works.

This problem seems to happen when I've not pushed any `local` commits to `origin/staging` for a while, I ran into this issue yesterday (about 12 hours ago).

I have a feeling the issue might be related to Redis and Horizon. I have to admit I don't know enough about the inner workings of this to know for sure.

- There's no errors in `daemon` log
- Commits are made in Statamic, they're just not always pushed to `origin/staging`

### Temporary fix

This is my process for *temporarily* fixing it

- `ssh` into server
- Check the origin path: `git remote -v`
- It's always set to `https`
- I change it to `ssh`
- `git push` now works in the terminal and changes made in Statamic are being pushed as expected.

I’m not sure what’s happening behind the scenes when I change the origin path, as when I use the deploy script in Ploi, the origin path reverts back to `https`. So perhaps the origin path isn’t related to this issue.

Something is forgetting or ignoring the credentials, as they work one day and not the other?

Open to suggestions!

**Comment by jesseleite** [comment]
Hmm, this is out of scope for Statamic, and more of a Github auth + Ploi thing, but maybe you can...

- Edit your deploy script to clone the repo using `ssh` instead of `http`?
- Or add a post-deploy hook to change it to `ssh` after the clone happens under the hood?
